src/main/java/fanny/Fanny.java

```diff
@@ -0,0 +1,105 @@
+package fanny;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import fanny.command.Command;
+import fanny.parser.Parser;
+import fanny.storage.Storage;
+import fanny.task.TaskList;
+import fanny.ui.Ui;
+
+/**
+ * The main class for the Fanny task management chatbot.
+ * Fanny is a text-based task manager chatbot that allows users
+ * to manage tasks, including to-dos, deadlines, and events.
+ */
+public class Fanny {
+
+    /** List of tasks */
+    private TaskList tasks;
+
+    /** Handles user interface interaction */
+    private Ui ui;
+
+    /** Handles the storing and loading of tasks */
+    private Storage storage;
+
+    private static final String FILEPATH = "data/fanny.txt";
+
+    public Fanny() {
+        this(FILEPATH);
+    }
+    /**
+     * Constructs a new Fanny object with the specified file path for storage.
+     *
+     * @param filePath The path of the file where tasks are stored.
+     */
+    public Fanny(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        tasks = new TaskList(storage);
+    }
+
+    /**
+     * Starts the Fanny chatbot and runs the main event loop.
+     */
+    public void run() {
+        ui.printHello();
+        boolean isExit = false;
```

Comment:
I think that isExit does not sound very boolean 😅 . I have a few suggestions. Perhaps you can use something like "hasExited", "shouldExit", or "isFinshed".

Reply:
Thank you for your comment. I believe shouldExit is indeed more apt in this case as the method specifies whether the program should exit after executing the command.